include/boost/gil/image_processing/numeric.hpp

```diff
@@ -295,6 +295,120 @@ inline void compute_hessian_entries(
     detail::convolve_2d(dy, sobel_y, ddyy);
 }
 
+namespace detail {
+enum class direction: std::size_t {
```

Comment:
Nitpick: `{` for types, functions, etc. apart from namespaces, should go in new line